packages/checkbox/src/mwc-checkbox.ts

```diff
@@ -0,0 +1,122 @@
+/**
+@license
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {html, FormElement, customElement, property, query, Foundation, Adapter, observer} from '@material/mwc-base/form-element';
+import {style} from './mwc-checkbox-css';
+import MDCCheckboxFoundation from '@material/checkbox/foundation';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'mwc-checkbox': Checkbox;
+  }
+}
+
+export interface CheckboxFoundation extends Foundation {
+  isChecked(): boolean;
+  setChecked(value: boolean): void;
+  setDisabled(disabled: boolean): void;
+  handleAnimationEnd(): void;
+  handleChange(): void
+}
+
+export declare var CheckboxFoundation: {
+  prototype: CheckboxFoundation;
+  new (adapter: Adapter): CheckboxFoundation;
+}
+
+@customElement('mwc-checkbox' as any)
```

Comment:
should this cast be here? (`as any`)

Reply:
Yes, we're working around https://github.com/Microsoft/TypeScript/issues/27171.
I've filed https://github.com/material-components/material-components-web-components/issues/114 to remove the cast when the upstream bug is fixed.

Reply:
ah good to know 👍